User

PranavBakre opened the pull request:
### TODO:
- [X] 1. Add admin routes for updating user and deleting user
- [x] 2. To implement is_active and admin approval
- [X] 3. To add expiry to access token and think of a logic for token refresh
~~- [ ] 4. To implement a class CRUDBase for CRUD operations( need to think if this is a good idea) to use instead of MongoEngine directly~~
- [x] 5. Response To remove or not to remove ??

PranavBakre committed:
Added get_current_admin method for getting the admin user

Fixed formatting with black

Signed-off-by: Pranav Bakre <pranavbakre@yahoo.in>

PranavBakre committed:
Added routes for delete and update self

Added response class for common response style

Tried to clean out User Schema

Formatting

Signed-off-by: Pranav Bakre <pranavbakre@yahoo.in>

PranavBakre committed:
get user by id

Signed-off-by: Pranav Bakre <pranavbakre@yahoo.in>

PranavBakre committed:
update user by id

Signed-off-by: Pranav Bakre <pranavbakre@yahoo.in>

#TODO: Create CRUD classes for operations. For Reason see update_user_by_id

prkprime commented:
why double quotes??????????????????????

PranavBakre commented:
@prkprime I did say i prefer double quotes

prkprime commented:
> @prkprime I did say i prefer double quotes

BC

PranavBakre commented:
@prkprime Dont curse on public repos

prkprime commented:
> @prkprime Dont curse on public repos

we hab 69 and 420 in our code xD
noone cares

PranavBakre committed:
delete user by Id

Signed-off-by: Pranav Bakre <pranavbakre@yahoo.in>

prkprime commented on app/api/api_v1/endpoints/user.py:
```diff
@@ -0,0 +1,89 @@
+    if update.is_admin is not None:
+        queried_user.is_admin = update.is_admin
+    queried_user.save()
+    return queried_user
```
this function allows admin to update normal user right?

prkprime commented on app/api/api_v1/endpoints/user.py:
```diff
@@ -0,0 +1,89 @@
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="Invalid Id")
+    queried_user = User.objects(id=id).first()
+    queried_user.delete()
+    return queried_user
```
admin can delete another admin
is that intended?

vigu184 replied:
Not needed. As we decide admin!

prkprime reviewed (commented):
so far everything looks fine
also @PranavBakre can write better python than me
me sed :(

PranavBakre committed:
User route and Token

changed route prefix for /user and started using Token class for sending access token and also added expiry to the fold

Signed-off-by: Pranav Bakre <pranavbakre@yahoo.in>

PranavBakre committed:
is_active check for authenticating user

Signed-off-by: Pranav Bakre <pranavbakre@yahoo.in>

#TODO: activate users

PranavBakre committed:
signup - is_active set to true if a normal user.

Signed-off-by: Pranav Bakre <pranavbakre@yahoo.in>

#TODO: Handle admin

rittmang commented:
Weeew

PranavBakre committed:
Admin Approval

Moved connect from models/__init__.py to server.py

Signed-off-by: Pranav Bakre <pranavbakre@yahoo.in>

PranavBakre committed:
Fixed a mistake where UserCreate also wanted is_active

added .vs/ in gitignore in case I become mad and open it in visual studio again

PranavBakre committed:
formatting

PranavBakre committed:
Comment responses and fixes

PranavBakre committed:
User Inactive should be 403, copied from rev-amp

PranavBakre committed:
Response class. To be used from now on

PranavBakre commented:
will do CRUDBase in new branch

rittmang commented:
@PranavBakre Go see Croods

rittmang reviewed (approved):
Bugre Ultra Pro Max

PranavBakre commented:
> Bugre Ultra Pro Max

rottmang

prkprime reviewed (approved):
noice
